src/providers/hover.ts

```diff
@@ -1,47 +1,60 @@
 import * as vscode from 'vscode';
 import * as state from '../state';
-import { nClient as nClient } from "../connector";
 import * as util from '../utilities';
 
 export default class HoverProvider implements vscode.HoverProvider {
     state: any;
+
     constructor() {
         this.state = state;
     }
-    formatDocString(nsname, arglist, doc) {
+
+    formatDocString(nsname: string, arglist: string, documentation: string) {
+
         let result = '';
+        // Format the namespace.
         if (nsname.length > 0 && nsname !== 'undefined') {
             result += '**' + nsname + '**  ';
             result += '\n';
         }
-
         // Format the different signatures for the fn
-        if (arglist & arglist.substring) {
+        if (arglist && arglist != "") {
             result += arglist.substring(0, arglist.length)
-                .replace(/\]/g, ']_')
-                .replace(/\[/g, '* _[');
+                .replace(/\]/g, ']')
+                .replace(/\[/g, '* [');
+            result += '\n\n';
             result += '\n\n';
         }
-
         // Format the actual docstring
-        if (doc) {
-            result += doc.replace(/\s\s+/g, ' ');
+        if (documentation && documentation != "") {
+            result += documentation.replace(/\s\s+/g, ' ');
             result += '  ';
         }
         return result.length > 0 ? result : "";
     }
 
     async provideHover(document, position, _) {
-        let text = util.getWordAtPosition(document, position);
-        if (this.state.deref().get('connected')) {
-            let client = util.getSession(util.getFileType(document));
+
+        if (util.getConnectedState()) {
```

Comment:
My knowledge of typescript is limited, but I'd suggest something like:
```
// in-ns util
function with-connection(f, ...) {
  function (args) {
     if (util.getConnectedState() {
       return  f(args);
    } else {
     return "Please connect to a REPL to retrieve information."
  }
}
```

Reply:
Also, I'd recommend decoupling this from the `hover` functionality itself.
It seems as `provideHover` "bottoms out" in 
`return new vscode.Hover(someString)`
So I'd suggest:

```
calculateHoverString(document, position) {
 const text = util.getWordAtPosition(document, position);
  const  ns = util.getNamespace(document);

  const client = util.getSession(util.getFileType(document));
  if(client) {
                 await util.loadFileIfNamespaceNotExist(document);
                 let res = await client.info(ns, text);
                 // I really don't not why result.arglists-str does 
                 // not work, but it leads to an compiler error.
                 if (res.ns && res.doc) {
                     return this.formatDocString(res.ns + "/" + res.name, res["arglists-str"] || [], res.doc)
                 }
                 if (res.ns) {
                     return this.formatDocString(res.ns + "/" + res.name, res["arglists-str"] || [], "No documentation available");
                 } else {
                     return this.formatDocString(text, "", "No information available");
                 }
             }
             return this.formatDocString(text, "", "No information available");
}

async provideHover(document, position, _) {
        const message = util.withConnection(calculateHoverString(document, position);
       return new score.Hover(message);
}
```

If you go in this direction, you could potentially unit-test `calculateHoverString`